sdk/azcore/core.go

```diff
@@ -125,16 +124,18 @@ func NewClient(clientName, moduleVersion string, plOpts runtime.PipelineOptions,
 	pl := runtime.NewPipeline(mod, moduleVersion, plOpts, options)
 
 	tr := options.TracingProvider.NewTracer(client, moduleVersion)
-	if tr.Enabled() && plOpts.TracingNamespace != "" {
-		tr.SetAttributes(tracing.Attribute{Key: "az.namespace", Value: plOpts.TracingNamespace})
+	if tr.Enabled() && plOpts.Tracing.Namespace != "" {
+		tr.SetAttributes(tracing.Attribute{Key: shared.TracingNamespaceAttrName, Value: plOpts.Tracing.Namespace})
 	}
 
+	// replace the vanilla tracer with this client-specific one
+	exported.SetTracer(&pl, tr)
```

Comment:
I admit that I don't totally love this part, but it seemed better than adding a new, exported, pipeline constructor that takes a `tracing.Tracer`.